.github/workflows/ci.yml

```
@@ -0,0 +1,112 @@
+name: CI
+on:
+  pull_request:
```

```suggestion
  pull_request:
      types: [ opened, synchronize, reopened ]
      branches:
        - 'main'
```

to avoid pull requests triggers into different branches (it would be overkill)

---

sorry but the point stay the same
```suggestion
  pull_request:
      types: [ opened, synchronize, reopened ]
      branches:
        - 'master'
```